Lib/test/test_typing.py

```diff
@@ -5719,14 +5719,10 @@ def athrow(self, typ, val=None, tb=None):
         self.assertNotIsInstance(g, G)
 
     def test_subclassing_subclasshook(self):
-
-        class Base(typing.Iterable):
+        class Base(abc.ABC):
```

Comment:
I think you've misunderstood the intent of this test, I'm afraid.

If the intent of this test was to check that custom `__subclasshook__` methods worked on subclasses of `abc.ABC`, then this test would be in `test_abc.py`, not `test_typing.py`. I believe the intent of this test is to check that subclasses of _`typing.Iterable`_ (and other `collections.abc` aliases in `typing.py`) can define custom `__subclasshook__` methods in the same way as the "original" classes in `collections.abc`, despite the fact that `typing.Iterable` isn't actually a class in the same way that `collections.abc.Iterable` is:

```pycon
>>> import typing
>>> type(typing.Iterable)
<class 'typing._SpecialGenericAlias'>
```

So, far from clarifying the intent of this test, I'm afraid this change turns it from being a useful test into a somewhat useless test.

Reply:
Also note that this test is found in a class called `CollectionsAbcTests`. It would be strange to have a test found in this class that didn't pertain specifically to an alias for a class from `collections.abc`.

Reply:
Thanks for clarifying @AlexWaygood, I've reverted that particular change.